src/client/src/orangehrmAdminPlugin/pages/socialMediaAuth/AddProvider.vue

```diff
@@ -0,0 +1,153 @@
+ * the GNU General Public License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+ -->
+
+<template>
+  <div class="orangehrm-background-container">
+    <div class="orangehrm-card-container">
+      <oxd-text tag="h6" class="orangehrm-main-title">
+        {{ $t('admin.add_provider') }}
+      </oxd-text>
+      <oxd-divider />
+      <oxd-form :loading="isLoading" @submit-valid="onSave">
+        <oxd-form-row>
+          <oxd-grid :cols="2" class="orangehrm-full-width-grid">
+            <oxd-grid-item>
+              <oxd-input-field
+                v-model="authProvider.name"
+                :rules="rules.name"
+                :label="$t('general.name')"
+                required
+              />
+            </oxd-grid-item>
+            <oxd-grid-item>
+              <oxd-input-field
+                v-model="authProvider.url"
+                :rules="rules.url"
+                :label="$t('admin.url')"
+                required
+              />
+            </oxd-grid-item>
+            <oxd-grid-item>
+              <oxd-input-field
+                v-model="authProvider.clientId"
+                :rules="rules.clientId"
+                :label="$t('admin.client_id')"
+                required
+              />
+            </oxd-grid-item>
+            <oxd-grid-item>
+              <oxd-input-field
+                v-model="authProvider.clientSecret"
+                :rules="rules.clientSecret"
+                :label="$t('admin.secret')"
+                required
+              />
+            </oxd-grid-item>
+          </oxd-grid>
+        </oxd-form-row>
+
+        <oxd-divider />
+        <oxd-form-actions>
+          <required-text />
+          <oxd-button
+            type="button"
+            display-type="ghost"
+            :label="$t('general.cancel')"
+            @click="onCancel"
+          />
+          <submit-button />
+        </oxd-form-actions>
+      </oxd-form>
+    </div>
+  </div>
+</template>
+
+<script>
+import {navigate} from '@ohrm/core/util/helper/navigation';
+import {APIService} from '@/core/util/services/api.service';
+import {
+  required,
+  shouldNotExceedCharLength,
+} from '@ohrm/core/util/validation/rules';
+import useServerValidation from '@/core/util/composable/useServerValidation';
+
+const initialAuthProvider = {
+  name: '',
+  url: '',
+  clientId: '',
+  clientSecret: '',
+  status: 1,
```

Comment:
hope this is optional parameter in BE API. better to set default value in BE and make it optional if not